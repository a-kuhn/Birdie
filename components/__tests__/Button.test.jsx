import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Button from "../Button";

describe("Button component", () => {
  it("renders correctly with default theme", () => {
    const { getByText, getByTestId, toJSON } = render(
      <Button label="Click me" onPress={() => {}} />
    );
    expect(getByText("Click me")).toBeTruthy();
    expect(getByTestId("default-theme")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with primary theme", () => {
    const { getByText, getByTestId, toJSON } = render(
      <Button label="Click me" theme="primary" onPress={() => {}} />
    );
    expect(getByText("Click me")).toBeTruthy();
    expect(getByTestId("primary-theme")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("calls the onPress function when pressed", () => {
    const onPressMock = jest.fn();
    const { getByText, toJSON } = render(
      <Button label="Click me" onPress={onPressMock} />
    );
    fireEvent.press(getByText("Click me"));
    expect(onPressMock).toHaveBeenCalled();
    expect(toJSON()).toMatchSnapshot();
  });
});
