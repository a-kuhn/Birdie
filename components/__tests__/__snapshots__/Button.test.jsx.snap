// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component calls the onPress function when pressed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": 68,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "padding": 3,
      "width": 320,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="default-theme"
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`Button component renders correctly with default theme 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": 68,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "padding": 3,
      "width": 320,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="default-theme"
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`Button component renders correctly with primary theme 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "height": 68,
        "justifyContent": "center",
        "marginHorizontal": 20,
        "padding": 3,
        "width": 320,
      },
      {
        "borderColor": "#ffd33d",
        "borderRadius": 18,
        "borderWidth": 4,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "backgroundColor": "#fff",
        },
      ]
    }
    testID="primary-theme"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
          },
          {
            "color": "#25292e",
          },
        ]
      }
    >
      Click me
    </Text>
  </View>
</View>
`;
